### PR TITLE
Support LIKE ESCAPE

### PR DESCRIPTION
**Pull request by ikrivosheev**
## PR Info

- Related:
  - https://github.com/SeaQL/sea-query/pull/94
  - https://github.com/SeaQL/sea-query/pull/318
  - https://github.com/SeaQL/sea-query/pull/352

## Adds

1. Move Like from `BinOp` into `SimpleExpr`
2. Add new struct: `LikeExpr`
3. Add new trait: `IntoLikeExpr`

### CODE REVIEW

**Comment by ikrivosheev** [comment]
CC @Sytten

**Comment by tyt2y3** [comment]
Thanks! I am about to make so modifications

File: src/value.rs [hunk 1]
```diff
@@ -48,6 +48,7 @@ pub enum Value {
     Float(Option<f32>),
     Double(Option<f64>),
     String(Option<Box<String>>),
+    Char(Option<char>),
```
Comment (tyt2y3) [inline src/value.rs:51]: Actually, I think we missed some impl in various drivers?

Comment (ikrivosheev) [reply src/value.rs:51]: I think we can remove this. This is my mistake...

**Pull request merged**